databricks/sdk/credentials_provider.py

```diff
@@ -664,6 +665,86 @@ def inner() -> Dict[str, str]:
     return inner
 
 
+class ModelServingAuthProvider():
+    _MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH = "/var/credentials-secret/model-dependencies-oauth-token"
+
+    @classmethod
+    def is_in_databricks_model_serving_environment(cls) -> bool:
+        """
+        Check if the code is running in Databricks Model Serving environment.
+        The environment variable set by Databricks when starting the serving container.
+        """
+        val = (
+            os.environ.get("IS_IN_DB_MODEL_SERVING_ENV")
```

Comment:
Could we test both of these env vars separately (maybe `pytest.mark.parametrize` the appropriate test to achieve this)? Having either should work right?